src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java

```diff
@@ -540,6 +542,80 @@ public void deleteSiteDependencies(String site) throws ServiceException {
+        List<CalculateDependenciesEntityTO> entities = new ArrayList<CalculateDependenciesEntityTO>();
+            Map<String, String> deps = calcualtePublishingDependencies(site, paths);
+            Map<String, List<Map<String, String>>> temp = new HashMap<String, List<Map<String,String>>>();
+            for (String p : paths) {
+                temp.put(p, new ArrayList<Map<String, String>>());
+            }
+            for (Map.Entry<String, String> d : deps.entrySet()) {
+                if (d.getKey() != d.getValue()) {
+                    List<Map<String, String>> ds = temp.get(d.getValue());
+                    ds.add(new HashMap<String, String>() {{
+                        put("item", d.getKey());
+                    }});
+                }
+            }
+            for (Map.Entry<String, List<Map<String,String >>> t : temp.entrySet()) {
+                CalculateDependenciesEntityTO calculateDependenciesEntityTO = new CalculateDependenciesEntityTO();
+                calculateDependenciesEntityTO.setItem(t.getKey());
+                calculateDependenciesEntityTO.setDependencies(t.getValue());
+                entities.add(calculateDependenciesEntityTO);
+            }
+
+        toRet.put("entities", entities);
+        return toRet;
+    }
+
+    private Map<String, String> calcualtePublishingDependencies(String site, List<String> paths)
+            throws SiteNotFoundException, ContentNotFoundException, ServiceException {
+        Set<String> toRet = new HashSet<String>();
+        Set<String> pathsParams = new HashSet<String>();
+
+        logger.debug("Get all publishing dependencies");
+        pathsParams.addAll(paths);
+        boolean exitCondition = false;
+        Map<String, String> ancestors = new HashMap<String, String>();
+        for (String p : paths) {
+            ancestors.put(p, p);
+        }
+        do {
+            List<Map<String, String>> deps = calculatePublishingDependenciesForListFromDB(site, pathsParams);
+            List<String> targetPaths = new ArrayList<String>();
+            for (Map<String, String> d : deps) {
+                String srcPath = d.get("source_path");
+                String targetPath = d.get("target_path");
+                if (!ancestors.keySet().contains(targetPath)) {
+                    if (!StringUtils.equals(targetPath, ancestors.get(srcPath))) {
+                        ancestors.put(targetPath, ancestors.get(srcPath));
+                    }
+                }
+                targetPaths.add(targetPath);
+            }
+            exitCondition = !toRet.addAll(targetPaths);
+            pathsParams.clear();
+            pathsParams.addAll(targetPaths);
+        } while (!exitCondition);
+
+        return ancestors;
+    }
+
+    private List<Map<String, String>> calculatePublishingDependenciesForListFromDB(String site, Set<String> paths) {
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put("site", site);
```

Comment:
Shouldn't these strings be in a constant somewhere (this line and the lines below)?